infrastructure/snyk_api/fake_api_client.go

```diff
@@ -28,12 +28,30 @@ type FakeApiClient struct {
 	LocalCodeEngine LocalCodeEngine
 	AutofixEnabled  bool
 	ApiError        *SnykApiError
+	Responses       map[string]any
 }
 
 var (
 	mutex = &sync.Mutex{}
 )
 
+func (f *FakeApiClient) SetResponse(method string, response any) {
+	if f.Responses == nil {
+		f.Responses = make(map[string]any)
+	}
+	f.Responses[method] = response
+}
+
+func (f *FakeApiClient) addCallForMethod(method string, args []any) {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	if f.Calls == nil {
+		f.Calls = make(map[string][][]any)
+	}
+	f.Calls[method] = append(f.Calls[method], args)
+}
+
```

Comment:
Added `addCallForMethod` to extend the testing for **Feature Flags**, without impacting existing consumers of the `FakeApiClient`. 

The current `addCall` is utilised across various tests and extending it to accommodate the new API call impact other consumers.

We could merge `addCall` with `addCallForMethod` into a single, more generic method in a future PR.